cowprotocol/cow-fi
# Refactor and add types

This PR does the following
- Fixes the issue where we were leaking these BIG JSON files into the client (👀 see Test section on how to check it out)
- Add strong and simpler types for both the token details (`TokenDetails`) and the token info in the list of tokens (`TokenInfo`)
- Now the parameters from the components are typed, and so it is the `getStaticProps` 🎉

Additionally, I moved for converting the logic to the `lib/tokens` (so we don't have part of the conversions/sorting in the `getStaticProps`, part in `lib`)

Lastly, I refactored a bit the components, for example, now the Listing has a `<ListItem />`


# Test

1. Open directly the Tokens page (list)

TEST 1: The pages loads. not JSON should be loaded
<img width="1167" alt="image" src="https://github.com/cowprotocol/cow-fi/assets/2352112/6e2b647c-c591-4ada-b1c0-cf2ea2a50b39">


2. Click on COW token

TEST 2: A simpler token JSON and only for COW will be fetched from the network:
<img width="1141" alt="image" src="https://github.com/cowprotocol/cow-fi/assets/2352112/81933e1c-5bc9-4076-bea9-128850600918">


3.  HARD REFRESH (the cow details page)

Test 3: The page loads. No JSON should be 


4.  Navigate to the list (click on the header menu)

Test 4: It should load the JSON of the list. This list should be way leaner (only the fields that are needed)

<img width="1218" alt="image" src="https://github.com/cowprotocol/cow-fi/assets/2352112/a3de0a3a-9cac-4e0c-913a-330be29dd4fe">
